### PR TITLE
added documentation for maintenance mode

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: cookbook/maintenance-mode.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    SetEnv SULU_MAINTENANCE true
+
+Configure Maintenance Mode
+--------------------------
+
+IP-Addreses:
+~~~~~~~~~~~~
+
+You may like to access your application while maintenance mode is active. Then you need to set $allowedIPs:
```
Comment (danrot) [inline cookbook/maintenance-mode.rst]: I would only write allowed IPs, the name of the variable can be seen in the code block.

File: cookbook/maintenance-mode.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
+Sulu Maintenance Mode
+=====================
+
+When you need to deploy a new version of your project on a production environment
+it is often necessary to disable your sulu-application and inform your customers
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: `s/customers/users`

File: cookbook/maintenance-mode.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+about it.
+
+Sulu maintenance mode displays a simple error page which can be easily customized.
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: `s/error/holding` perhaps, but its not an error page.

File: cookbook/maintenance-mode.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Create Maintenance Mode
+-----------------------
+
+To create a maintenance page, you first need to create a maintenance.php file:
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Put filenames in double backticks:

````
You first need to create a ``maintenance.php`` file
````

File: cookbook/maintenance-mode.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+.. code-block:: bash
+
+    $ cp app/maintenance.php.dist app/maintenance.php
+
+Then you need to set the env variable SULU_MAINTENANCE to true.
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: constant should be in double backticks

Comment (dantleech) [inline cookbook/maintenance-mode.rst]: `s/env/environment`

File: cookbook/maintenance-mode.rst [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+-----------------------
+
+To create a maintenance page, you first need to create a maintenance.php file:
+
+.. code-block:: bash
+
+    $ cp app/maintenance.php.dist app/maintenance.php
+
+Then you need to set the env variable SULU_MAINTENANCE to true. 
+E.g.: in your .htaccess file (for apache)
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: `.htaccess` in double backticks

Comment (dantleech) [inline cookbook/maintenance-mode.rst]: `s/E.g./For example, in your`

File: cookbook/maintenance-mode.rst [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+ 
+.. code-block:: php
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: this is not php. use `apache`

File: cookbook/maintenance-mode.rst (hunk 1 above)
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: yes, `You need to set the allowed IPs`

File: cookbook/maintenance-mode.rst [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+--------------------------
+
+IP-Addreses:
+~~~~~~~~~~~~
+
+You may like to access your application while maintenance mode is active. Then you need to set $allowedIPs:
+
+.. code-block:: php
+
+    $allowedIPs = array(
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Need to add the `<?php` tag for this to display properly.

File: cookbook/maintenance-mode.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        '127.0.0.1'
+    );
+
+Translations:
+~~~~~~~~~~~~~
+You can define translations for your template as follows:
+
+.. code-block:: php
+
+    $translations = array(
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Add `<?php` tag

File: cookbook/maintenance-mode.rst [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+.. code-block:: php
+
+    $allowedIPs = array(
+        '127.0.0.1'
+    );
+
+Translations:
+~~~~~~~~~~~~~
+You can define translations for your template as follows:
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Always a blank line after a section heading.

File: cookbook/maintenance-mode.rst [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+    $translations = array(
+        'en' => array(
+            'title' => 'Maintenance',
+            'heading' => 'The page is currently down for maintenance',
+            'description' => 'Sorry for any inconvenience caused. Please try again shortly.',
+        ),
+    );
+    
+Default locale:
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Titles should not have `:`

File: cookbook/maintenance-mode.rst [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+~~~~~~~~~~~~~~~
+
+By default, maintenance.php is automatically detecting your browsers language. If no translation for this language exists
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: filenames (`maintenance.php`) in double backticks

Comment (dantleech) [inline cookbook/maintenance-mode.rst]: This line is too long, it should not exceed ~80 chars.

File: cookbook/maintenance-mode.rst [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+the default locale is being used. By default this is english:
+
+.. code-block:: php
+
+    define('DEFAULT_LOCALE', 'en');
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Add `<?php` tag

File: cookbook/maintenance-mode.rst [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+            'heading' => 'The page is currently down for maintenance',
+            'description' => 'Sorry for any inconvenience caused. Please try again shortly.',
+        ),
+    );
+    
+Default locale:
+~~~~~~~~~~~~~~~
+
+By default, maintenance.php is automatically detecting your browsers language. If no translation for this language exists
+the default locale is being used. By default this is english:
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: English needs to be Capitalized.

File: cookbook/maintenance-mode.rst [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+~~~~~~~~~~~~~~
+
+By default, ``maintenance.php`` is automatically detecting your browsers language. If no translation for this language 
+exists the default locale is being used. By default this is English:
+
+.. code-block:: php
+
+    <?php
+    define('DEFAULT_LOCALE', 'en');
+    ?>
```
Comment (danrot) [inline cookbook/maintenance-mode.rst]: We don't use the closing tags

Comment (turbo-ele) [inline cookbook/maintenance-mode.rst]: :(

File: cookbook/maintenance-mode.rst [hunk 16]
```diff
@@ -0,0 +1,70 @@
+Sulu Maintenance Mode
+=====================
+
+When you need to deploy a new version of your project on a production environment
+it is often necessary to disable your sulu-application and inform your users
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: Sulu application -- @danrot  ?

File: cookbook/maintenance-mode.rst [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+For example, in your ``.htaccess`` file (for apache)
+ 
+.. code-block:: apache
+
+    SetEnv SULU_MAINTENANCE true
+
+Configure Maintenance Mode
+--------------------------
+
+IP-Addreses
```
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: No hyphenation in IP Addresses, `Allowed IP addresses` probably better title.

File: cookbook/maintenance-mode.rst (hunk 15 above)
Comment (dantleech) [inline cookbook/maintenance-mode.rst]: yeah, closing tags are not necessary

File: cookbook/maintenance-mode.rst (hunk 16 above)
Comment (danrot) [inline cookbook/maintenance-mode.rst]: I guess so

**Pull request merged**